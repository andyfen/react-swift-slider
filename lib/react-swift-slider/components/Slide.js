'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true,
});
exports['default'] = void 0;

var _react = _interopRequireDefault(require('react'));

var _styledComponents = _interopRequireWildcard(require('styled-components'));

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj;
  } else {
    var newObj = {};
    if (obj != null) {
      for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
          var desc =
            Object.defineProperty && Object.getOwnPropertyDescriptor
              ? Object.getOwnPropertyDescriptor(obj, key)
              : {};
          if (desc.get || desc.set) {
            Object.defineProperty(newObj, key, desc);
          } else {
            newObj[key] = obj[key];
          }
        }
      }
    }
    newObj['default'] = obj;
    return newObj;
  }
}

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

function _typeof(obj) {
  if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
    _typeof = function _typeof(obj) {
      return typeof obj;
    };
  } else {
    _typeof = function _typeof(obj) {
      return obj &&
        typeof Symbol === 'function' &&
        obj.constructor === Symbol &&
        obj !== Symbol.prototype
        ? 'symbol'
        : typeof obj;
    };
  }
  return _typeof(obj);
}

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError('Cannot call a class as a function');
  }
}

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if ('value' in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

function _possibleConstructorReturn(self, call) {
  if (call && (_typeof(call) === 'object' || typeof call === 'function')) {
    return call;
  }
  return _assertThisInitialized(self);
}

function _assertThisInitialized(self) {
  if (self === void 0) {
    throw new ReferenceError(
      "this hasn't been initialised - super() hasn't been called",
    );
  }
  return self;
}

function _getPrototypeOf(o) {
  _getPrototypeOf = Object.setPrototypeOf
    ? Object.getPrototypeOf
    : function _getPrototypeOf(o) {
        return o.__proto__ || Object.getPrototypeOf(o);
      };
  return _getPrototypeOf(o);
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== 'function' && superClass !== null) {
    throw new TypeError('Super expression must either be null or a function');
  }
  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: { value: subClass, writable: true, configurable: true },
  });
  if (superClass) _setPrototypeOf(subClass, superClass);
}

function _setPrototypeOf(o, p) {
  _setPrototypeOf =
    Object.setPrototypeOf ||
    function _setPrototypeOf(o, p) {
      o.__proto__ = p;
      return o;
    };
  return _setPrototypeOf(o, p);
}

function _templateObject3() {
  var data = _taggedTemplateLiteral([
    '\n  background-position: center;\n  background-size: cover;\n  position: absolute;\n  left: 0;\n  top: 0;\n  width: 100%;\n  height: 100%;\n  z-index: ',
    ';\n  visibility: ',
    ';\n  transition: visibility 0.3s;\n  animation: ',
    " 0.8s;\n  background-image: url('",
    "');\n  will-change: opacity;\n  will-change: visibility;\n  will-change: z-index;\n",
  ]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral([
    '\n  from {\n    opacity: 1;\n  }\n  \n  to {\n    opacity: 0;\n  }\n',
  ]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral([
    '\n  from {\n    opacity: 0;\n  }\n  \n  to {\n    opacity: 1;\n  }\n',
  ]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }
  return Object.freeze(
    Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } }),
  );
}

var fadeIn = (0, _styledComponents.keyframes)(_templateObject());
var fadeOut = (0, _styledComponents.keyframes)(_templateObject2());

var SwiftSliderSlide = _styledComponents['default'].li(
  _templateObject3(),
  function(props) {
    return props.active ? '2' : '0';
  },
  function(props) {
    return props.active ? 'visible' : 'hidden';
  },
  function(props) {
    return props.active ? fadeIn : fadeOut;
  },
  function(props) {
    return props.backgroundImage;
  },
);

var Slide =
  /*#__PURE__*/
  (function(_React$Component) {
    _inherits(Slide, _React$Component);

    function Slide() {
      _classCallCheck(this, Slide);

      return _possibleConstructorReturn(
        this,
        _getPrototypeOf(Slide).apply(this, arguments),
      );
    }

    _createClass(Slide, [
      {
        key: 'render',
        value: function render() {
          var _this$props = this.props,
            src = _this$props.src,
            active = _this$props.active;
          return _react['default'].createElement(SwiftSliderSlide, {
            backgroundImage: src,
            active: active,
          });
        },
      },
    ]);

    return Slide;
  })(_react['default'].Component);

exports['default'] = Slide;
