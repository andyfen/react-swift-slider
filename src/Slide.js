import React from 'react';
import { StyleSheet, css } from 'aphrodite';

class Slide extends React.Component {
  render(){
    const { src, active } = this.props;
    return(
      <li
        className={css(styles.slide, active && styles.active)}
        style={{backgroundImage: 'url(' + src + ')'}}/>
    )
  }
}
export default Slide;

const styles = StyleSheet.create({
  slide: {
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    position: 'absolute',
    left: 0,
    top: 0,
    width: '100%',
    height: '100%',
    opacity: 0,
    zIndex: 1,
    transition: 'opacity 1s'
  },
  active: {
    opacity: 1,
    zIndex: 2,
  },
  image: {
    maxWidth: '300px',
    marginRight: '2rem',
  }
})
