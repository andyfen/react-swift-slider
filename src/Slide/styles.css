
.slide {
    background-position: center;
    background-size: cover;
    position: absolute;
    left: 0px;
    top: 0px;
    width: 100%;
    height: 100%;
    opacity: 0;
    z-index: 1;

    -webkit-transition: opacity 1s;
    -moz-transition: opacity 1s;
    -o-transition: opacity 1s;
    transition: opacity 1s;
}
.slide:nth-of-type(1) {
    background: red;
}
.slide:nth-of-type(2) {
    background: orange;
}
.slide:nth-of-type(3) {
    background: green;
}
.slide:nth-of-type(4) {
    background: blue;
}
.slide:nth-of-type(5) {
    background: purple;
}

.active {
    opacity: 1;
    z-index: 2;
}

.image {
  max-width: 300px;
  margin-right: 2rem;
}
