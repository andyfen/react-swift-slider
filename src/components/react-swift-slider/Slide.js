import React, { Component } from "react";
import PropTypes from "prop-types";
import "./assets/sass/react-swift-slide.css";

class Slide extends Component {
  render() {
    const { src, active } = this.props;
    const selectedClass = active
      ? "swift-slider-slide swift-slider-active"
      : "swift-slider-slide";
    return (
      <li
        className={selectedClass}
        style={{ backgroundImage: "url(" + src + ")" }}
      />
    );
  }
}

Slide.propTypes = {
  src: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
};

export default Slide;
