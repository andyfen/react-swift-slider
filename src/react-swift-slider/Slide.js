import React from 'react';
import type { SlideProps } from './types/Slide.Types';

export default class Slide extends React.Component {
  render() {
    const { src, active } = this.props;
    const selectedClass = active
      ? 'swift-slider-slide swift-slider-active'
      : 'swift-slider-slide';
    return (
      <li
        className={selectedClass}
        style={{ backgroundImage: 'url(' + src + ')' }}
      />
    );
  }
}
