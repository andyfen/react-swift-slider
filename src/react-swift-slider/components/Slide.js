import React from 'react';
import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  
  to {
    opacity: 1;
  }
`;

const fadeOut = keyframes`
  from {
    opacity: 1;
  }
  
  to {
    opacity: 0;
  }
`;

const SwiftSliderSlide = styled.li`
  background-position: center;
  background-size: cover;
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: ${props => (props.active ? '2' : '0')};
  visibility: ${props => (props.active ? 'visible' : 'hidden')};
  transition: visibility 0.3s;
  animation: ${props => (props.active ? fadeIn : fadeOut)} 0.8s;
  background-image: url('${props => props.backgroundImage}');
  will-change: opacity;
  will-change: visibility;
  will-change: z-index;
`;

export default function Slide({ src, active }) {
  return <SwiftSliderSlide backgroundImage={src} active={active} />;
}
