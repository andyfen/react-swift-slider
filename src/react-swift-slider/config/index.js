export const DIRECTION = {
  prev: 'prev',
  next: 'next',
};

export const CSS_OVERRIDE = {
  swiftSliderContainerClass: 'swift-slider-container',
  swiftSliderSlidesClass: 'swift-slider-slides',
  swiftSliderDotsClass: 'swift-slider-dots',
  swiftSliderDotClass: 'swift-slider-dot',
  swiftSliderActiveDotClass: 'swift-slider-active-dot',
  swiftSliderSlideClass: 'swift-slider-slide',
  swiftSliderActiveSlideClass: 'swift-slider-active-slide',
  swiftSliderPreviousBtnClass: 'swift-slider-prev-btn',
  swiftSliderNextBtnClass: 'swift-slider-next-btn',
};

export const CSS_OVERRIDE_EXTERNAL = {
  ...Object.keys(CSS_OVERRIDE)
    .map((key) => ({
      [key]: `.${CSS_OVERRIDE[key]}`,
    }))
    .reduce((prev, cur) => ({ ...prev, ...cur }), {}),
};

export const DEFAULT_PROPS = {
  data: [],
  height: 450,
  activeDotColor: '#e8784e',
  interval: 5000,
  dotColor: '#909192',
  showDots: true,
  enableNextAndPrev: true,
};
