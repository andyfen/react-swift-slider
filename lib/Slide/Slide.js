import React from 'react';
import classNames from 'classnames/bind';

import styles from './styles.css';
const cx = classNames.bind(styles);

class Slide extends React.Component {
  render(){
    const slideClasses = cx({
      slide: true,
      active: this.props.active,
    });
    const { src } = this.props;
    return(
      <li className={slideClasses} style={{backgroundImage: 'url(' + src + ')'}}>
        
      </li>
    )
  }
}
export default Slide;
