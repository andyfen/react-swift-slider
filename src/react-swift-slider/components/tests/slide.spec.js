/**
 * Created by Muzikayise Flynn Buthelezi (zuluCoda) on 2018/07/27.
 * Copyright mfbproject.co.za - muzi@mfbproject.co.za
 * Copyright zulucoda - mfbproject
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Slide from '../Slide';
import { CSS_OVERRIDE_EXTERNAL } from '../../config';

describe('React Swift Slider: Slide - Unit Test', () => {
  afterEach(cleanup);

  describe('when active', () => {
    it('should render Slide with active css properties', () => {
      const props = {
        active: true,
        src: 'some src url',
      };

      const { container } = render(
        <Slide src={props.src} active={props.active} />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
  describe('when NOT active', () => {
    it('should render Slide without active css properties', () => {
      const props = {
        active: false,
        src: 'some src url',
      };

      const { container } = render(
        <Slide src={props.src} active={props.active} />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('css override handles', () => {});
});
