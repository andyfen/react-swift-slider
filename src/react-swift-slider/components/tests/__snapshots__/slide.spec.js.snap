// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Swift Slider: Slide - Unit Test when NOT active should render Slide without active css properties 1`] = `
<li
  class="sc-bdfBwQ ecRqZV"
/>
`;

exports[`React Swift Slider: Slide - Unit Test when active should render Slide with active css properties 1`] = `
<li
  class="sc-bdfBwQ rGDLu"
/>
`;
